src/integrations/GoogleAds/utils.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * if trackEvents are not enabled then do not send events (return false)
+ * if trackEvents are enabled but event filtering is not enabled then send all events
+ * if trackEvents are enabled and event filtering is also enabled then check if event is configured to event list or not
+ * if non of above is true then do not send event at all (return false)
+ * @param {*} eventName
+ * @param {*} trackEvents
+ * @param {*} enableFiltering
+ * @param {*} events
+ * @returns
+ */
+function shouldSendEvent(eventName, trackEvents, enableFiltering, events) {
+  if (!trackEvents) {
+    return false;
+  }
+
+  if (trackEvents && !enableFiltering) {
+    return true;
+  }
+
+  if (trackEvents && enableFiltering) {
+    const eventNames = [];
+    events.forEach((event) => {
+      if (event.eventName !== '') {
+        eventNames.push(event.eventName);
+      }
+    });
+    return eventNames.includes(eventName.trim());
```

Comment:
```suggestion
    events.forEach((event) => {
      if(event.eventName === eventName.trim())
        return true;
    });
```

Reply:
@ItsSudip we don't return from forEach loop generally, we will process inside loop and then return final value after loop